doc/design/adding-vertical-pod-autoscaler.md

```diff
@@ -0,0 +1,44 @@
+# Adding Vertical Pod Autoscaler 
+
+## Description
+
+OLM supports users including `VerticalPodAutoscaler` (VPA) objects in their bundle alongside their operator manifests. `VerticalPodAutoscalers`
+objects are used to configure the VerticalPodAutoscaler controller to dynamically allocate resources to pods based on their usage of CPU, memory, 
+and other custom metrics. VPAs allow for more efficient use of cluster resources as pod resource needs are continually evaluated and adjusted by the VPA controller.
+For more info, see the docs at https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler. 
+
+## Caveats
+
+`VerticalPodAutoscaler` objects watch a controller reference, such as deployment, to find a collection of pods to resize. Be sure to pass
+the appropriate reference to your operator or operands depending on which you would like the VPA to watch. 
+
+The VerticalPodAutoscaler controller must be enabled and active in the cluster for the VPA objects included in the bundle to have an effect. 
+Alternatively, the installing operator could also add the VPA as a required API to ensure the VPA operator is present in the cluster.
+
+The VPA will continually terminate pods and adjust the resource limits as needed - be sure your application is tolerant of restarts
+before including a VPA alongside it. 
+
+Note: at this time it is not recommended for the VPA to run alongside the HorizontalPodAutoscaler (HPA) on the same set of pods. 
+VPA can however be used with an HPA that is configured to use either external or custom metrics. 
+
+## Technical Details
+
+VPA yaml manifests can be placed in the bundle alongside existing manifests in the `/manifests` directory. The VPA manifest will be present
```

Comment:
Can we get into a situation where we have contradictory VPAs installed? Are there really no sticky situations we can get into when dealing with VPAs during an upgrade; e.g. an old VPA prevents a new CSV's pods from scaling up?

Is the idea to pass the buck on these questions until we have a generic way to specify install behavior?

Reply:
I think so. The VPA spec is extremely straightforward - what object is being autoscaled, and the update strategy employed by the VPA for that object. I looked briefly into the update strategy as a way of mitigating risk but the reality is it's hard to picture the ways in which the VPA can affect upgrades. It may be fine, or it may blow up. 

I think we will get more clarity as we get users actually using them in the bundles